### PR TITLE
Remove all V2 metadata string keys, use V3 metadata enum keys instead

### PR DESCRIPTION
**Pull request by mqliang**
## Description

Remove all V2 metadata string keys (except the "Exception" one, which is used in V2 to construct exception metadata keys), use V3 metadata enum keys instead.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes
If you have tagged this as either backward-incompat or release-notes,
you MUST add text here that you would like to see appear in release notes of the
next release.

If you have a series of commits adding or enabling a feature, then
add this section only in final commit that marks the feature completed.
Refer to earlier release notes to see examples of text

## Documentation
If you have introduced a new feature or configuration, please add it to the documentation as well.
See https://docs.pinot.apache.org/developers/developers-and-contributors/update-document

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java [hunk 1]
```diff
@@ -30,22 +30,8 @@
  * Data table is used to transfer data from server to broker.
  */
 public interface DataTable {
+  // TODO: remove this when we stop supporting DataTable V2.
   String EXCEPTION_METADATA_KEY = "Exception";
-  String NUM_DOCS_SCANNED_METADATA_KEY = "numDocsScanned";
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java:34]: Before removing these, please make sure and double-check that these are the exact strings (with no typos) used in `MetadataKeys`. I think some of these are returned in the results metadata by broker? We should take care not to break compat.

Comment (mqliang) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java:34]: I have made a double check. There is no typos.

**Comment by mqliang** [comment]
test failed due to flaky issue, close&reopen to trigger a re-run

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java [hunk 2]
```diff
@@ -55,15 +56,12 @@
...
-  private static final Logger LOGGER = LoggerFactory.getLogger(BrokerReduceService.class);
-
-  // brw -> Shorthand for broker reduce worker threads.
-  private static final String REDUCE_THREAD_NAME_FORMAT = "brw-%d";
-
   // Set the reducer priority higher than NORM but lower than MAX, because if a query is complete
   // we want to deserialize and return response as soon. This is the same as server side 'pqr' threads.
   protected static final int QUERY_RUNNER_THREAD_PRIORITY = 7;
-
+  private static final Logger LOGGER = LoggerFactory.getLogger(BrokerReduceService.class);
```
Comment (jackjlli) [inline pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java:62]: Nit: it'd be good to keep these static constants on the top.

**Comment by Jackie-Jiang** [comment]
There is an unrelated file: `pinot-integration-tests/examples/batch/baseballStats/segments/baseballStats_OFFLINE_0.tar.gz` (generated during the quick-start)

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java (hunk 2 above)
Comment (mqliang) [reply pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java:62]: This change is performed by Intellj reformat, so I's like to not change it back.

**Comment by mqliang** [comment]
@Jackie-Jiang unrelated file removed. PTAL.

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java [hunk 3]
```diff
@@ -86,6 +84,31 @@ public BrokerReduceService(PinotConfiguration config) {
     _reduceExecutorService = Executors.newFixedThreadPool(numThreadsInExecutorService, reduceThreadFactory);
   }
 
+  private static void updateAlias(QueryContext queryContext, BrokerResponseNative brokerResponseNative) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java:87]: Let's not use IDE to rearrange the code (there is a checkbox during reformat). We don't usually follow that rule, and that is not really help the readability.

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise. Please resolve the conflict

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java (hunk 3 above)
Comment (mqliang) [reply pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java]: done

**Review by jackjlli — APPROVED** [review]

**Comment by mqliang** [comment]
unit-test failed due to flaky issue. close&re-open to trigger a re-run

**Comment by mqliang** [comment]
unit-test time out due to case `org.apache.pinot.plugin.stream.kafka20.KafkaPartitionLevelConsumerTest` failed:
```
2021-04-13T05:29:42.3106099Z [INFO] Running org.apache.pinot.plugin.stream.kafka20.KafkaPartitionLevelConsumerTest
2021-04-13T05:31:53.4390023Z 05:29:44.845 NetworkClient - [AdminClient clientId=adminclient-1] Connection to node -1 could not be established. Broker may not be available.
2021-04-13T05:31:53.4392386Z 05:29:44.966 NetworkClient - [AdminClient clientId=adminclient-1] Connection to node -1 could not be established. Broker may not be available.
2021-04-13T05:31:53.4394513Z 05:29:45.168 NetworkClient - [AdminClient clientId=adminclient-1] Connection to node -1 could not be established. Broker may not be available.
2021-04-13T05:31:53.4396529Z 05:29:45.470 NetworkClient - [AdminClient clientId=adminclient-1] Connection to node -1 could not be established. Broker may not be available.
2021-04-13T05:31:53.4398884Z 05:29:45.569 BrokerMetadataCheckpoint - No meta.properties file under dir /tmp/mini-kafka-cluster4639957122637414558/broker-1156490705878060507/meta.properties
2021-04-13T05:31:53.4402528Z 05:29:45.972 NetworkClient - [AdminClient clientId=adminclient-1] Connection to node -1 could not be established. Broker may not be available.
2021-04-13T05:31:53.4406209Z 05:29:46.567 BrokerMetadataCheckpoint - No meta.properties file under dir /tmp/mini-kafka-cluster4639957122637414558/broker-1156490705878060507/meta.properties
2021-04-13T05:31:53.4478982Z 05:29:48.446 RequestSendThread - [RequestSendThread controllerId=0] Controller 0's connection to broker fv-az204-998.internal.cloudapp.net:34677 (id: 0 rack: null) was unsuccessful
2021-04-13T05:31:53.4481642Z java.net.SocketTimeoutException: Failed to connect within 1500 ms
2021-04-13T05:31:53.4484383Z 	at kafka.controller.RequestSendThread.brokerReady(ControllerChannelManager.scala:280) [kafka_2.11-2.0.0.jar:?]
2021-04-13T05:31:53.4487209Z 	at kafka.controller.RequestSendThread.doWork(ControllerChannelManager.scala:233) [kafka_2.11-2.0.0.jar:?]
2021-04-13T05:31:53.4489496Z 	at kafka.utils.ShutdownableThread.run(ShutdownableThread.scala:82) [kafka_2.11-2.0.0.jar:?]
```
I can successfully run the test case locally. Looks like a flaky issue. close&reopen to trigger a re-run.

**Comment by siddharthteotia** [comment]
@mqliang can you please rebase and force push?

**Comment by mqliang** [comment]
@siddharthteotia > can you please rebase and force push?
Done.

**Pull request merged**